src/Tuura/Plato/Translation.hs

```diff
@@ -6,7 +6,13 @@ import Data.List.Extra
 import Tuura.Concept.Circuit.Basic
 import Tuura.Concept.Circuit.Derived
 
-data ValidationResult a = Valid | Invalid [a] [a] [a] [a] deriving Eq
+data ValidationResult a = Valid | Invalid [ValidationError a] deriving Eq
+
+data ValidationError a = UnusedSignal [a]
```

Comment:
I think `UnusedSignal a` is more precise. If there are two undefined signals, you simply do `Invalid [UnusedSignal a, UnusedSignal b]`. With you current approach there are multiple ways to achieve the same thing, e.g. `Invalid [UnusedSignal [a, b]]`.